### PR TITLE
Changed IDE0059 codefix title in pattern matching

### PR DESCRIPTION
**Pull request by DoctorKrolic**
Fixes: https://github.com/dotnet/roslyn/issues/40002

### CODE REVIEW

File: src/Analyzers/Core/CodeFixes/RemoveUnusedParametersAndValues/AbstractRemoveUnusedValuesCodeFixProvider.cs [hunk 1]
```diff
@@ -162,6 +162,13 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
                             title = CodeFixesResources.Remove_redundant_assignment;
                         }
 
+                        // Also we want to show "Remove redundant assignment" title in pattern matching, e.g.
+                        // if (obj is SomeType someType) <-- "someType" will be fully removed here
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/CodeFixes/RemoveUnusedParametersAndValues/AbstractRemoveUnusedValuesCodeFixProvider.cs:166]: how are we sure of that?  Can you link to the respective fixer code that always does this?  Thanks!

Comment (DoctorKrolic) [reply src/Analyzers/Core/CodeFixes/RemoveUnusedParametersAndValues/AbstractRemoveUnusedValuesCodeFixProvider.cs:166]: The code is buried somewhere in the analizer. I am not familiar with how operation-based analizers work. Hoever, by looking through test, I didn't find any test, where expected behaviour is replacement of original variable with `_`

**Comment by DoctorKrolic** [comment]
@CyrusNajmabadi please review

**Review by CyrusNajmabadi — APPROVED** [review]
Thanks!

**Comment by DoctorKrolic** [comment]
@CyrusNajmabadi I checked failing integration tests and found out, that `IsUnusedLocalDiagnostic` will be reported for all unused local nodes, e.g.:
```cs
class Program
{
  public static void Main()
  {
    var x = new Program();
  }
}
```
`x` will have `IsUnusedLocalDiagnostic` tag, but codefix will produce this code:
```cs
class Program
{
  public static void Main()
  {
    var _ = new Program();
  }
}
```
So the decision of whether should node be removed or changed to discard is fully implemented in codefixe's code. This method replaces all declaration patterns with type declarations:
https://github.com/dotnet/roslyn/blob/afe689ab1a902bde4af3c6c5e419f898b9347fb3/src/Analyzers/CSharp/CodeFixes/RemoveUnusedParametersAndValues/CSharpRemoveUnusedValuesCodeFixProvider.cs#L79-L93
So in cases like
```cs
var a = "str";
if (a is object obj)
{
}
```
declaration pattern `object obj` will be changed to just type declaration `object`

Considering this fact, I reverted logic of PR back to just checking for declaration pattern when deciding what codefix title to show

**Comment by DoctorKrolic** [comment]
@CyrusNajmabadi please have a look

**Comment by DoctorKrolic** [comment]
@CyrusNajmabadi PTAL

**Comment by CyrusNajmabadi** [comment]
Thanks :)

**Pull request merged**